internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	for _, change := range params.Changes {
+		URI := string(change.URI)
+		dir, err := os.Stat(URI)
+		if err != nil {
+			return fmt.Errorf("Unable to update %s. Failed to open directory, error: %w", URI, err)
+		}
+
+		filename := path.Base(URI)
```

Comment:
```suggestion
		uri := string(change.URI)
		fi, err := os.Stat(uri)
		if err != nil {
			// TODO: raise warning via window/showMessage
			continue
		}
		if fi.IsDir() {
			// TODO: raise warning via window/showMessage
			continue
		}

		filename := path.Base(uri)
```
Usually in Go internal variables are lowercased/camelcased.

Reply:
Thanks for the tip, I've been following [these](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) guidelines from Go so far which states the following:
```
Words in names that are initialisms or acronyms (e.g. "URL" or "NATO")
have a consistent case. For example, "URL" should appear as "URL" or "url"
(as in "urlPony", or "URLPony"), never as "Url".
```

But I agree to your point on having that distinction for internal variables and otherwise. Thanks

Reply:
> "URL" should appear as "URL" or "url" (as in "urlPony", or "URLPony"), never as "Url"

That seems right and not in conflict :) it's just that `URL` is more often used e.g. for constants or exported variables, `url` is what I'd use for internal variable.

Reply:
I completely understand, I was just trying to understand the intent/reasoning. This actually makes more sense as I said and I'd personally be using this rule going forward. Thanks for taking the time to clear this up, really appreciate it :)